controller/node_controller.go

```diff
@@ -73,6 +75,18 @@ type NodeController struct {
 
 type TopologyLabelsChecker func(kubeClient clientset.Interface, vers string) (bool, error)
 
+// Ref: https://github.com/longhorn/longhorn/issues/7931
+type defectiveKernelRange struct {
+	broken string
+	fixed  string
+}
+
+// TODO: Keep an up-to-date bad kernel list in upgrade-responder and pull the latest info from there.
+var knownDefectiveKernels = []defectiveKernelRange{
+	{"5.15.0-94", "5.15.0-100"},
+	{"6.5.6", "6.5.9"},
+}
```

Comment:
I am wondering if, in addition to @PhanLe1010's suggestion to have somewhere we can pull these from dynamically (which can be handled in another ticket), we should have this behavior dynamically configurable on the cluster.

- Let's say we get it wrong. Maybe some distro is built on `v6.5.6` but it backports a fix such that `v6.5.6-1-fixedit` works.
- Let's say a user is using a broken kernel, but definitely does not intend on ever using NFSv4 or RWX in general.

Now the user sees perpetual red lights for a situation they don't care about.

Should we have a setting like:

- `disable-kernel-version-check`, or
- `ignore-kernel-version-check-for` which can include a list of kernels to ignore, or
- `known-defective-kernels` which includes the above list by default, but can be edited by the user as desired?

Reply:
Yes, I think an override setting would be useful (or maybe more than one, since it might be handy to have both "disable" and "override-list" capabilities.)  The latter, if we publish it somewhere that is easily copy/paste-able (the KB article itself, perhaps) also gives a handy solution to the air-gapped case that doesn't require a software upgrade to get new hard-coded values.

Reply:
That being said, at the moment, this gives us a handy troubleshooting item.  Would it be appropriate to push the other aspects into a follow-on ticket?

Reply:
> have somewhere we can pull these from dynamically (which can be handled in another ticket), we should have this behavior dynamically configurable on the cluster.

Do you mean making it become a setting like `Stable Longhorn Versions`? Then the upgrade checker can help pull the info from the server, in which we can modify the kernel versions dynamically.